### PR TITLE
Improve gallery, text boxes, footer and other stuff

### PR DESCRIPTION
**Pull request by TobiGr**
This PR

- changes the gallery framework

- adds some texts to the gallery

- changes the text boxes' background

- changes some other styles

- improves the use of bootstrap

- adds a link to the CCC presentation

- removes unused code

- includes other small changes

### CODE REVIEW

**Comment by theScrabi** [comment]
All right thanks :)

File: gallery.css [hunk 1]
```diff
@@ -130,20 +144,21 @@
         min-height: 100vh;
         width: calc(var(--items-mobile) * 100%);
     }
-    .gallery-element {
-        min-height: 100vh;
+    .gallery-controls {
+        width: calc(100% / var(--items-mobile));
```
Comment (TheAssassin) [inline gallery.css]: CSS variables are not supported very well yet, except for the most modern browsers (released no earlier than one year ago). See http://caniuse.com/#feat=css-variables. This leads to problems on older mobile phones.
You should replace these with fixed values, as you did almost everywhere else. You can also convert this to LESS/SCSS if you want to, too, to be able to use variables and other neat features like color calculation.

File: gallery.css [hunk 2]
```diff
@@ -27,6 +27,19 @@
...
+.gallery-element.gallery-desktop-item {
+    display: initial;
+}
+.gallery-element[name^=item] {
+    background: #858585; /* Fallback for browsers that do not support gradients */
+    background: -webkit-linear-gradient(left,rgba(133,133,133,1), rgba(170,170,170,1),rgba(133,133,133,1)); /*Safari 5.1-6*/
+    background: -o-linear-gradient(right,rgba(133,133,133,1), rgba(170,170,170,1),rgba(133,133,133,1)); /*Opera 11.1-12*/
+    background: -moz-linear-gradient(right,rgba(133,133,133,1), rgba(170,170,170,1),rgba(133,133,133,1)); /*Fx 3.6-15*/
+    background: linear-gradient(to right, rgba(133,133,133,1), rgba(170,170,170,1),rgba(133,133,133,1)); /*Standard*/
+}
```
Comment (TheAssassin) [inline gallery.css]: This gradient is almost invisible on larger screens. You should think about increasing the contrast, for example by changing the color stops to be closer to the edges.

File: index.html [hunk 3]
```diff
@@ -127,19 +127,10 @@ <h1 class="hidden-xs">NewPipe</h1>
...
-                        <a href="#item4" class="unselected"><div></div></a>
-                        <a href="#item5" class="unselected"><div></div></a>
-                        <a href="#item6" class="unselected"><div></div></a>
-                        <a href="#item7" class="unselected"><div></div></a>
-                    </div>
+                    <div class="controls-bottom"></div>
                 </div>
                 
-                <div class="gallery-element" name="item1">
+                <div class="gallery-element gallery-desktop-item" name="item1" style="background: transparent">
```
Comment (TheAssassin) [inline index.html]: Inline CSS is an unforgivable sin. Please move the code to your CSS file.

File: js/gallery.js [hunk 4]
```diff
@@ -79,33 +86,51 @@ function galleryScroll(x){
...
         $('.gallery').css('marginLeft', -(current - 2) * width + 'px');
         $('.gallery-controls').css('marginLeft', (current - 2) * width + 'px');
     }
 }
 
+function galleryAddControls(){
+    $('.controls-bottom').empty();
+    for(var i = 1; i < items + 1; i++){
+        if(i == 1){
+            $('.controls-bottom').append('<a href="#item'+ i +'" class="selected"><div></div></a>');
```
Comment (TheAssassin) [inline js/gallery.js]: I won't merge this kind of code. If you need templating, you shouldn't use jQuery for this, you need a templating engine.

File: js/gallery.js [hunk 5]
```diff
@@ -79,33 +86,51 @@ function galleryScroll(x){
...
+    }
+}
+
 function galleryAddStyle(){
-    var nd = $('.gallery-element[name*="item"]').length;
-    var nm = nd - 1;
+    var nd = $('.gallery-desktop-item').length + 1;
+    var nm = $('.gallery-mobile-item').length + 1;
     document.querySelector('style').textContent +=             
     ".gallery { width: calc(100% * " + nd + " );} .gallery-element, .gallery-controls { width: calc(100% / " + nd + " ); } @media (max-width:766px) { .gallery { width: calc(100% * " + nm + " );} .gallery-element, .gallery-controls { width: calc(100% / " + nm + " ); } }";
```
Comment (TheAssassin) [inline js/gallery.js]: I overlooked this code obviously, we are going to have to change that.

File: gallery.css [hunk 6]
```diff
@@ -130,20 +144,21 @@
...
     }
-    .gallery-element {
-        min-height: 100vh;
+    .gallery-controls {
+        width: calc(100% / var(--items-mobile));
     }
     .gallery-controls .controls-bottom {
         display: none;
     }
-    .gallery-element[name="item1"]{
```
Comment (TheAssassin) [inline gallery.css]: Good move to change this, this is really bad style.

**Comment by TheAssassin** [comment]
I would prefer to see an approach without all this custom code. There's tons of ready-to-use open-source galleries out there, so I'd rather like to see one of them being used, because that appears to be more maintainable. Otherwise, I'd suggest you to switch to a framework like AngularJS (I prefer v1), because that'll greatly reduce the amount of code you have to write.

**Comment by TobiGr** [comment]
At the moment I am waiting for the final design. Until I do not have it, I won't make any big changes to the gallery and the header. And then I will replace the parallax.js by a pure CSS parallax which doesn't need that much resources. 
And yes it's better to not use CSS variables. And there are some other bad styles in the CSS files but I have no time to fix them soon. 
So please be patient - there will be good results in the end ;)

**Comment by TheAssassin** [comment]
Great. We all appreciate your hard work of course. I'm looking forward to your updates.

**Comment by theScrabi** [comment]
To be honest, I have no idea about web programming, so I can't tell you guys what you should do write and what you should do wrong.

I trust both of you :)

**Comment by theScrabi** [comment]
@TobiGr In other words would It be worth merging even if refactoring would be required in the future, and do you think that that would be a problem?
 @TheAssassin because if not, I'd merge it.

**Comment by TheAssassin** [comment]
@theScrabi I guess we could merge it, but as the site is not ready to deploy yet (placeholders etc.), I'd wait for @TobiGr to improve his commits. In my experience, refactoring has to be done in the first place to not increase complexity too much. Otherwise we'd have to remember that in some way, and usually nobody cares about that in a few weeks or months any more.
I trust @TobiGr to finish these features when he has some spare time. This is not too urgent yet.

**Comment by TobiGr** [comment]
All issues should be resolved.

**Comment by TobiGr** [comment]
Firefox for Android does not pauses the video when it's not displayed.

**Comment by TheAssassin** [comment]
Interesting... what does it mean that Firefox for Android does not pause the video? Higher CPU usage in the background?

**Comment by TobiGr** [comment]
Yes. Without pausing the video we have a higher CPU usage.
Same problem with Firefox Desktop.

**Comment by TheAssassin** [comment]
Interesting (and weird) behavior. They should document that better. I'm curious how you found out about it...

**Comment by TobiGr** [comment]
There was a notification by Firefox that a video was playing on my tablet.

**Comment by TobiGr** [comment]
Do you think it's necessary to move the JS script to an external file?
I'm not sure if we'll need it somewhere else and it's pretty short too.

Otherwise, I don't see any more things I'd change for this PR.

**Comment by TobiGr** [comment]
@TheAssassin I'm sorry for making this PR so messy and big...
I have prepared some other changes. As you can see in [#10](https://github.com/TeamNewPipe/website/issues/10#issuecomment-290934865). I changed the text boxes including some changes to bootstrap use and deleting unused stuff. As you said, you didn't mind that is getting messy. 
Should I push them here (the changes have nothing to do with the gallery...)?

**Comment by TheAssassin** [comment]
Yeah, just push them into this PR.

Your PR cannot be merged automatically by GitHub any more due to some conflict, so I'll do it manually as soon as you pushed the rest of your commits.

**Comment by TobiGr** [comment]
That's it. Here you go :)

**Comment by TheAssassin** [comment]
JFYI: Going to review (and hopefully merge) this PR tonight.

**Comment by TobiGr** [comment]
Perfect 👍 
At the moment there are two links to the CCC presentation. I will remove my link later and think about a good solution so yours fits to the used styles.
Because I have little time now I will focus on it later. But I think you can merge it anyway.

**Comment by TheAssassin** [comment]
I've merged it, but would like to note that the Nexus 6 frame is missing on a slide, please fix that soon-ish.

**Comment by TobiGr** [comment]
I think the screenshots look more modern without the Nexus frame. I am thinking of showing all screenshots without a frame. (Except for the animation)

**Comment by TheAssassin** [comment]
Oh okay, a reasonable design decision. We can compare it when you're done.

**Pull request merged**